jenkinsTerraform/main.tf

```diff
@@ -0,0 +1,90 @@
+#################### Configure AWS provider ####################
+#################### based on https://registry.terraform.io/providers/hashicorp/aws/latest/docs ####################
+provider "aws" {
+  region     = var.region
+  access_key = var.access_key
+  secret_key = var.secret_key
+  # profile = "Admin"
+}
+
+#################### Configure Jenkins infrastructure ####################
+
+#################### Instance #1 - Jenkins Manager ####################
+resource "aws_instance" "deployment8_jenkins_manager" {
+  ami                         = var.ami
+  instance_type               = var.deployment8_instance_type
+  vpc_security_group_ids      = [aws_security_group.jenkins.id]
+  key_name                    = var.key_name
+  associate_public_ip_address = true
+
+  subnet_id = file("../initTerraform/main.tf")
+  user_data = file("jenkins-manager-installs.sh")
+  tags = {
+    Name = "jenkins-manager-east"
+  }
+}
+
+#################### Instance #2 - Jenkins Agent01 ####################
+resource "aws_instance" "deployment8_jenkins_agent01" {
+  ami                         = var.ami
+  instance_type               = var.deployment8_instance_type
+  vpc_security_group_ids      = [aws_security_group.jenkins.id]
+  key_name                    = var.key_name
+  associate_public_ip_address = true
+
+  subnet_id = file("../initTerraform/main.tf")
+  user_data = file("jenkins-agent01-installs.sh")
+  tags = {
+    Name = "jenkins-manager-east"
+  }
+
+}
+
+#################### Instance #3 - Jenkins Agent02 ####################
+resource "aws_instance" "deployment8_jenkins_agent02" {
+  ami                         = var.ami
+  instance_type               = var.deployment8_instance_type
+  vpc_security_group_ids      = [aws_security_group.jenkins.id]
+  key_name                    = var.key_name
+  associate_public_ip_address = true
+
+  subnet_id = file("../initTerraform/main.tf")
```

Comment:
I think this should be placed in the default vpc subnet because the vpc created in the initTerraform folder won't be created till after we log into Jenkins and run the pipeline.